### ISSUE TITLE
Narrator doesn't announce the "checked", "unchecked" when pressing Space keyboard, after using mouse to change check/uncheck state of toolStripButton

### ISSUE THREAD

**Issue by Amy-Li03** [dotnet/winforms#9461]
### .NET version

.NET 8.0.100-preview.7.23360.13

### Did it work in .NET Framework?

No

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

No

### Issue description

When verifying https://github.com/dotnet/winforms/issues/9199, find a new issue, it only reproduces on this specific scenario: using mouse to change check/uncheck state of toolStripButton, then pressing Space keyboard, Narrator cannot announce "checked", "unchecked".
![NewIssue](https://github.com/dotnet/winforms/assets/62929087/4eb8fe08-45d3-4528-a366-3aaf509931e1)

### Steps to reproduce

1. Create a Winforms .Net application with ToolStrip
2. Add a ToolStripButton, set its CheckOnClick propterty to True
3. Build and run
4. Launch Narrator, use mouse to change check/uncheck state of toolStripButton
5. Press Space keyboard to change check/uncheck state of toolStripButton
[TestApp.zip](https://github.com/dotnet/winforms/files/12011675/TestApp.zip)

**Comment by Tanya-Solyanik** [comment]
@v-elnovikova  - could you please take a look if this is a quick fix and share your ideas in this bug. This is not ahigh priority issue because the repro uses mouse and screen reader users usually use only the keyboard, so if this takes a long time, we'll deprioritize it.

**Comment by Tanya-Solyanik** [comment]
Thank you for the investigation and the office example, makes sense. @merriemcgaw, I'm comfortable with closing this issue,  thoughts?

**Comment by merriemcgaw** [comment]
I am also comfortable with this not being addressed.
